service/storage_service.cc

```diff
@@ -342,6 +342,10 @@ future<> storage_service::join_token_ring(cdc::generation_service& cdc_gen_servi
         // Check if the node is already removed from the cluster
         auto local_host_id = _db.local().get_config().host_id;
         auto my_ip = get_broadcast_address();
+        if (_gossiper.is_quarantined(local_host_id)) {
```

Comment:
Note: unfortunately it doesn't prevent "zombies" from rejoining if they were running all this time  when we removed/replaced them (only partitioned away).

Still, it's an improvement over current situations.

But does it actually **fix** #11217 / #5523 as the commit message suggests? If we want to be precise - not entirely.

Reply:
@11217 in particular does talk about network partitioning so you're right, Ref'fing it would be more accurate.
Although, on the process side of things, I think we'd be better off with opening a new issue for fencing off a network-partitioned node when it's expelled from the cluster, since this would be achieved later using the raft infrastructure and I guess extending rpc to support fencing this way.

Reply:
Added comment about this case in v3
and change "Fixes" in commitlog tags and description of this issue to "Refs"